### PR TITLE
chore(plugins/auth-backend): migrate from passport-saml package to @node-saml/passport-saml

### PR DESCRIPTION
**Pull request by secustor**
## Hey, I just made a Pull Request!

Migrates to the renamed package `passport-saml` to `@node-saml/passport-saml`

I couldn't test this as I have no SAML setup available tough this should work without issue. 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by secustor** [comment]
I have exposed the necessary configuration options, which are mentioned in the predecessor.

File: .changeset/tasty-oranges-rescue.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+'@backstage/plugin-auth-backend': minor
+---
+
+migrate from 'passport-saml' to '@node-saml/passport-saml'  
+This comes with breaking changes to config options:
+
+- `audience` is no mandatory
+- `wantAuthnResponseSigned` is now exposed and defaults to `true`
+- `wantAssertionsSigned` is now exposed and defaults to `true`
```
Comment (Rugvip) [inline .changeset/tasty-oranges-rescue.md:10]:
```suggestion
**BREAKING**: The `saml` provider has been migrated from `passport-saml` to `@node-saml/passport-saml`.
  
This comes with breaking changes to config options:

- `audience` is now mandatory
- `wantAuthnResponseSigned` is now exposed and defaults to `true`
- `wantAssertionsSigned` is now exposed and defaults to `true`
```

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 👍 🎉

File: .changeset/tasty-oranges-rescue.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+'@backstage/plugin-auth-backend': minor
+---
+
+**BREAKING**: The `saml` provider has been migrated from `passport-saml` to `@node-saml/passport-saml`.
+  
+This comes with breaking changes to config options:
+
+- `audience` is now mandatory
+- `wantAuthnResponseSigned` is now exposed and defaults to `true`
+- `wantAssertionsSigned` is now exposed and defaults to `true`
```
Comment (Rugvip) [inline .changeset/tasty-oranges-rescue.md:11]:
```suggestion
**BREAKING**: The `saml` provider has been migrated from `passport-saml` to `@node-saml/passport-saml`.

This comes with breaking changes to config options:

- `audience` is now mandatory
- `wantAuthnResponseSigned` is now exposed and defaults to `true`
- `wantAssertionsSigned` is now exposed and defaults to `true`

```

File: .changeset/tasty-oranges-rescue.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+---
+
+**BREAKING**: The `saml` provider has been migrated from `passport-saml` to `@node-saml/passport-saml`.
+
+This comes with breaking changes to config options:
+
+- `audience` is now mandatory
+- `wantAuthnResponseSigned` is now exposed and defaults to `true`
+- `wantAssertionsSigned` is now exposed and defaults to `true`
+
```
Comment (Rugvip) [inline .changeset/tasty-oranges-rescue.md:12]:
```suggestion
---
'@backstage/plugin-auth-backend': minor
---

**BREAKING**: The `saml` provider has been migrated from `passport-saml` to `@node-saml/passport-saml`.

This comes with breaking changes to config options:

- `audience` is now mandatory
- `wantAuthnResponseSigned` is now exposed and defaults to `true`
- `wantAssertionsSigned` is now exposed and defaults to `true`
```

**Comment by Rugvip** [comment]
Shipping early so that we can have it included in today's release, test failures are unrelated

**Pull request merged**